R/diffResult.R

```diff
@@ -0,0 +1,194 @@
+                ncol_base = ncol(base),
+                ncol_comp = ncol(comp)
+            )
+        },
+        
+        add_checkResult = function(CR){
+            self$checks[[CR$name]] <- CR
+        },
+        
+        generate_summary = function(){
+            
+            if(is.null(self$keys)){
+                keys <- "  None Provided  "
+            } else {
+                keys <- self$keys
+            }
+            
+            keysum <- data.frame(
+                "Variable" = keys,
+                stringsAsFactors = FALSE
+            )
+            
+            class_base <- as_cropped_char( paste0(self$meta$class_base, collapse = ", "))
+            class_comp <- as_cropped_char( paste0(self$meta$class_comp, collapse = ", "))
+            
+            
+            dfsum <-  data.frame(
+                "Summary" = c("Number of rows", "Number of columns", "Class"),
+                "Base" = c( self$meta$nrow_base, self$meta$ncol_base, class_base),
+                "Compare" = c( self$meta$nrow_comp, self$meta$ncol_comp, class_comp),
+                stringsAsFactors = FALSE
+            )
+            
+            results <- data.table(
+                Name = character(0),
+                Result = character(0)
+            )
+            for( i in self$checks){
+                results <- rbind(results, data.table(
+                    Name = i$name,
+                    Result = i$result
+                ))
+            }
+            
+            self$check_summary = results
+            
+            if(self$result == "Passed"){
+                line <- "All checks have passed"
+            } else {
+                line <- "Not all checks have passed, details are provided below"
+            }
+            
+            self$summary <- display$new(
+                title = sprintf("Comparison of Base vs Compare"),
+                body = list(
+                    "Dataset Summary",
+                    dfsum,
+                    "",
+                    "Listing of Keys",
+                    keysum,
+                    "",
+                    "Check Summary",
+                    results,
+                    "",
+                    line,
+                    ""
+                )
+            )
+        }
+    )
+)
+
+
+
+
+#' print.diffResult
+#' 
+#' Print method for a diffdf result object (diffResult). Can be rendered
+#' into different output formats and saved into a report file.
+#' 
+#' @param x diffResult Object
+#' @param type Output format. Options = "ascii" (default) or "html"
+#' @param rowlimit How many rows of a display dataset should be shown (default = 10)
+#' @param file Location and name of a text file to output the results to. Setting to NULL will cause no file to be produced.
+#' @param display Whether to print the rendered output in the R console
+#' @param ... Additional arguments (not used)
+#' 
+#' @export
+print.diffResult <- function(
+    x, 
+    type = "ascii", 
```

Comment:
use matchargs() to list possible options

Reply:
Sorry not sure I quite understand what you mean here

Reply:
If you do type = c("ascii", "html") and then in code have type <- match.arg(type) that forces the user to select one of the possible options. It also makes clear in documentation what possible options they can choose.

Reply:
I'm not sure about this, I get the appeal but match.arg supports argument completion meaning the following would be valid which I hate. 

```
print(x , type = "ht")
```

Reply:
I mean, this is fairly standard for R packages though... I think in general this kind of argument completion is useful and it's how R works. For example, R autocompletes argument names in functions, and even in $. E.g. if I do X <- data.frame(arg = "cake"); X$a; This actually works!

I agree that it's probably not the best behaviour in the world, but its something the user is used to, and it gives the big benefit of allowing us to pre-specify the allowed arguments, which is really nice from a documentation point of view. If you hate the argument completion enough we could always make our own version of match.arg which strips that out